Framework/Algorithms/test/AppendSpectraTest.h

```diff
@@ -25,210 +27,429 @@ class AppendSpectraTest : public CxxTest::TestSuite {
+    // the values
+    const auto inputAxis1 = inputWS1->getAxis(1);
+    const auto inputAxis2 = inputWS2->getAxis(1);
+    const auto outputAxis = outputWS->getAxis(1);
+
+    const auto ws1len = inputWS1->getNumberHistograms();
+
+    for (size_t i = 0; i < outputWS->getNumberHistograms() / 2; ++i) {
+      // check that all labels are the same
+      TS_ASSERT_EQUALS(inputAxis1->getValue(0), outputAxis->getValue(i));
+      // this will check the labels for the second workspace that is
+      // appended at position starting from the length of the first workspace
+      TS_ASSERT_EQUALS(inputAxis2->getValue(0),
+                       outputAxis->getValue(i + ws1len));
     }
+  }
 
-    auto ws1Log = new TimeSeriesProperty<std::string>("aLog");
-    ws1Log->addValue(DateAndTime("2014-06-19T16:40:00"), "Hello");
-    ws1->mutableRun().addLogData(ws1Log);
-
-    auto ws2Log = new TimeSeriesProperty<std::string>("aLog");
-    ws2Log->addValue(DateAndTime("2014-06-19T16:40:10"), "World");
-    ws2->mutableRun().addLogData(ws2Log);
-
-    AnalysisDataService::Instance().addOrReplace(ws1Name, ws1);
-    AnalysisDataService::Instance().addOrReplace(ws2Name, ws2);
-
-    AppendSpectra alg;
-    alg.initialize();
-    TS_ASSERT_THROWS_NOTHING(alg.setPropertyValue("InputWorkspace1", ws1Name));
-    TS_ASSERT_THROWS_NOTHING(alg.setPropertyValue("InputWorkspace2", ws2Name));
-    TS_ASSERT_THROWS_NOTHING(alg.setPropertyValue("OutputWorkspace", ws1Name));
-    if (combineLogs)
-      alg.setProperty("MergeLogs", true);
-    TS_ASSERT_THROWS_NOTHING(alg.execute();)
-    TS_ASSERT(alg.isExecuted());
-
+private:
+  void doTestAppendSpectraWithWorkspaces(const std::string &inputWorkspace1,
+                                         const std::string &inputWorkspace2,
+                                         const std::string &outputWorkspace) {
+    auto appendSpectra =
+        Mantid::API::FrameworkManager::Instance().createAlgorithm(
+            "AppendSpectra");
+    TS_ASSERT_THROWS_NOTHING(
+        appendSpectra->setProperty("InputWorkspace1", inputWorkspace1));
     TS_ASSERT_THROWS_NOTHING(
-        out = boost::dynamic_pointer_cast<MatrixWorkspace>(
-            AnalysisDataService::Instance().retrieve(ws1Name));)
-    TS_ASSERT(out);
-    if (!out)
-      return;
-
-    TS_ASSERT_EQUALS(out->getNumberHistograms(), 15);
-    TS_ASSERT_EQUALS(out->blocksize(), numBins);
-
-    for (size_t wi = 0; wi < out->getNumberHistograms(); wi++) {
-      TS_ASSERT_EQUALS(out->getSpectrum(wi).getSpectrumNo(), specnum_t(wi));
-      TS_ASSERT(!out->getSpectrum(wi).getDetectorIDs().empty());
-      for (size_t x = 0; x < out->blocksize(); x++)
-        TS_ASSERT_DELTA(out->readY(wi)[x], 2.0, 1e-5);
+        appendSpectra->setProperty("InputWorkspace2", inputWorkspace2));
+    TS_ASSERT_THROWS_NOTHING(
+        appendSpectra->setProperty("OutputWorkspace", outputWorkspace));
+    TS_ASSERT_THROWS_NOTHING(appendSpectra->execute());
```

Comment:
To my understanding, call to execute() never throws, unless you do `appendSpectra->setRethrows(true);`.